src/entry.rs

```diff
@@ -16,32 +16,34 @@ pub enum Entry<'a, K: IntKey, V: 'a> {
 impl<'a, K: IntKey, V> Entry<'a, K, V> {
     #[inline]
     pub(crate) fn new(key: K, int_map: &'a mut IntMap<K, V>) -> Self {
-        let indices = Self::indices(key, int_map);
+        let (cache_ix, vals_ix) = Self::indices(key, int_map);
 
-        match indices {
-            Some((cache_ix, vals_ix)) => Entry::Occupied(OccupiedEntry {
+        match vals_ix {
+            Some(vals_ix) => Entry::Occupied(OccupiedEntry {
                 vals_ix,
                 vals: &mut int_map.cache[cache_ix],
                 count: &mut int_map.count,
             }),
-            None => Entry::Vacant(VacantEntry { key, int_map }),
+            None => Entry::Vacant(VacantEntry {
+                key,
+                cache_ix,
+                int_map,
+            }),
         }
     }
 
-    fn indices(key: K, int_map: &IntMap<K, V>) -> Option<(usize, usize)> {
+    fn indices(key: K, int_map: &IntMap<K, V>) -> (usize, Option<usize>) {
         if int_map.is_empty() {
-            return None;
+            return (0, None);
```

Comment:
Returning `0` only works because we'll increase the cache before inserting the element. This wasn't always true. It would be nice to have comment here that documents this invariant. Something like `// Returning 0 is okay because we'll increase the cache and recalculate the index`